File: mario.c
#include <cs50.h>
#include <stdio.h>

void print_row(int spaces, int bricks);
void print_symbol(int num, string symbol);
void check_valid_height(int height);

int main(void)
{
    int height;
    do
    {
        height = get_int("Height: ");
        check_valid_height(height);
    }
    while (height < 1 || height > 8);

    for (int row = 1; row <= height; row++)
    {
        int spaces = height - row;
        int bricks = height - spaces;
        print_row(spaces, bricks);
    }
}

void check_valid_height(int height)
{
    if (height < 1)
    {
        printf("Min height is 1\n");
    }

    if (height > 8)
    {
        printf("Max height is 8\n");
    }
}

void print_symbol(int num, string symbol)
{
    for (int i = 0; i < num; i++)
    {
        printf("%s", symbol);
    }
}

void print_row(int spaces, int bricks)
{
    print_symbol(spaces, " ");
    print_symbol(bricks, "#");
    printf("  ");
    print_symbol(bricks, "#");
    printf("\n");
}

File: mario.py
from cs50 import get_int


def print_symbol(num: int, symbol: str) -> None:
    """
    Prints the symbol provided num times
    """
    for i in range(num):
        print(f"{symbol}", end="")


def print_row(spaces: int, bricks: int) -> None:
    """
    Prints a double row of blocks and spaces
    """
    symbols = {'space': ' ', 'brick': '#'}

    print_symbol(spaces, symbols['space'])
    print_symbol(bricks, symbols['brick'])
    print(f"{symbols['space']}" * 2, end="")
    print_symbol(bricks, symbols['brick'])
    print()


def main() -> None:
    """
    Prints a double Super Mario row of blocks
    """
    while True:
        height = get_int("Height: ")
        if height > 0 and height < 9:
            break
        print("Please, enter a valid hight between 1 and 8")

    for row in range(1, height + 1):
        spaces = height - row
        bricks = height - spaces
        print_row(spaces, bricks)


main()
